components/server/src/code-sync/code-sync-service.ts

```diff
@@ -303,4 +292,25 @@ export class CodeSyncService {
         return router;
     }
 
+    protected async deleteObjects(userId: string, resourceKey: ServerResource, revs: string[]) {
+        revs.forEach(rev => this.db.deleteResource(userId, resourceKey, rev, async (rev: string) => {
+            const obj = toObjectName(resourceKey, rev);
+            try {
+                const request = new DeleteRequest();
+                request.setOwnerId(userId);
+                request.setExact(obj);
+                const blobsClient = this.blobsProvider.getDefault();
+                await util.promisify<DeleteRequest, DeleteResponse>(blobsClient.delete.bind(blobsClient))(request);
+            } catch (err) {
+                if (err.code === status.NOT_FOUND) {
+                    return;
+                }
+                log.error({ userId }, 'code sync: failed to delete obj', err, { object: obj });
+                throw err;
+            }
+        }).catch(err => {
+            log.error({ userId }, 'code sync: failed to delete', err);
```

Comment:
Remove this catch and handle it on caller function?

Reply:
Any of them failed is ok, we should continue to delete other items

Reply:
If you catch here then this function will never throw and catch below :arrow_down: will never execute